dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -1036,4 +1037,38 @@ private void checkNotHosted(Tool tool) {
             throw new CustomWebApplicationException("Cannot modify hosted entries this way", HttpStatus.SC_BAD_REQUEST);
         }
     }
+
+
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/{toolId}/zip/{tagId}")
+    @ApiOperation(value = "Download a ZIP file of a tool and all associated files.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) })
+    @Produces("application/zip")
+    public Response getToolZip(@ApiParam(hidden = true) @Auth Optional<User> user, @ApiParam(value = "toolId", required = true) @PathParam("toolId") Long toolId,
+                                   @ApiParam(value = "tagId", required = true) @PathParam("tagId") Long tagId) {
+        Tool tool;
+        if (user.isPresent()) {
+            tool = toolDAO.findById(toolId);
+            checkEntry(tool);
+            checkUser(user.get(), tool);
+        } else {
+            tool = toolDAO.findPublishedById(toolId);
+            checkEntry(tool);
+        }
+
+        Tag tag = tool.getTags().stream().filter(innertag -> innertag.getId() == tagId).findFirst().orElseThrow(() -> new CustomWebApplicationException("Could not find tag", HttpStatus.SC_NOT_FOUND));
+        Set<SourceFile> sourceFiles = tag.getSourceFiles();
+        if (sourceFiles == null || sourceFiles.size() == 0) {
+            return null;
+        }
+
+        String fileName = tool.getToolPath().replaceAll("/", "-") + ".zip";
+
+        File returnZipFile = downloadAsZip(sourceFiles, fileName);
```

Comment:
This will never get deleted, will it?

Googled around and there's this: https://stackoverflow.com/questions/26930127/how-to-delete-file-after-rest-response